homeassistant/components/modem_callerid/config_flow.py

```diff
@@ -0,0 +1,76 @@
+"""Config flow for Modem Caller ID integration."""
+from phone_modem import PhoneModem, exceptions
+import serial
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.const import CONF_DEVICE, CONF_NAME
+
+from . import _LOGGER
+from .const import DEFAULT_DEVICE, DEFAULT_NAME
+from .const import DOMAIN  # pylint:disable=unused-import
+
+DATA_SCHEMA = vol.Schema({"name": str, "device": str})
+
+
+class PhoneModemFlowHandler(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for Phone Modem."""
+
+    async def async_step_user(self, user_input=None):
+        """Handle a flow initiated by the user."""
+        errors = {}
+
+        if user_input is not None:
+            name = user_input[CONF_NAME]
+            device = user_input[CONF_DEVICE]
+
+            if await self._async_endpoint_existed(device):
```

Comment:
This can be simplified using existing helpers for unique ID:
```
                await self.async_set_unique_id(device)
                self._abort_if_unique_id_configured()
```

Reply:
Relatedly the service calls don't seem to have the ability to differentiate between different modems so should this be guarded against via single instance abort, instead?

```
if self.hass.config_entries.async_entries(DOMAIN):
            return self.async_abort(reason="single_instance_allowed")
```

But maybe set unique ID to keep the option open for supporting multiple in the future?

Reply:
How can you tell that the service calls can only do one modem? I don't have 2 modems to determine if 2 configurations are possible. However, its very unlikely that current users would try to use 2 in a single HA instance.

Reply:
My thinking for that is because the `hangup_call` and `reject_call` functions never test for an entity_id or similar. Unless async_register is doing some magic to filter the service call, but I don't see it given any entity specific information to be able to do that.

This might be something you could make a mock test for and see what happens?

Reply:
> This can be simplified using existing helpers for unique ID:
> 
> ```
>                 await self.async_set_unique_id(device)
>                 self._abort_if_unique_id_configured()
> ```

According to @frenck , this only works for new configurations. Older configs do not have unique id implemented. The service calls in my pypi package currently only do one modem. Changes would have to be made to support multiple modems. Could anyone really be using more than one modem?

Reply:
@joncar Looks like I was able to make more than one modem supported now. In my testing, I found that the handup call service did not actually do anything so it has been removed.

Reply:
RE: new configurations. Isn't config flow and creation of config entries brand new in this PR? What older configurations exist in the wild?

RE: multiple modems. It does seem unlikely in a home. But also who doesn't build their office's phone system using Home Assistant? ;)